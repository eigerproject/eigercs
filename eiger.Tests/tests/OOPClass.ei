class Person
	let name
	let surname

	func new(name, surname)
		this.name = name
		this.surname = surname
	end

	func Introduction()
		emitln("I am " + this.name + " " + surname + "!")
	end

	func Greet(other)
		emitln(name + " greeted " + other.name + "!")
	end
end

let px = Person("Name1", "Surname1")
let py = Person("Name2", "Surname2")

px.Introduction()
py.Introduction()

px.Greet(py)