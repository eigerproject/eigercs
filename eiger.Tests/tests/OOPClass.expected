I am Name1 Surname1!
I am Name2 Surname2!
Name1 greeted Name2!